scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -1083,7 +1082,14 @@ object TreeSyntax {
         if (isTemplateEmpty) s()
         else {
           val pearly = if (!t.early.isEmpty) s("{ ", r(t.early, "; "), " } with ") else s()
-          val pparents = w(r(t.inits, " with "), " ", !t.inits.isEmpty && !isBodyEmpty)
+          val suffix = if (t.inits.nonEmpty && !isBodyEmpty) " " else ""
+          val canHaveExtend = t.parent.exists(p =>
```

Comment:
what about `t.parent.exists(!_.is[Term.NewAnonymous])` (or `(_.is[Defn])`)?

Reply:
Changed to `t.parent.exists(p => p.isNot[Defn.Given] && p.isNot[Term.NewAnonymous])`